noir-projects/aztec-nr/aztec/src/encrypted_logs/incoming_body.nr

```diff
@@ -1,51 +1,27 @@
 use crate::note::{note_interface::NoteInterface};
 use dep::protocol_types::{grumpkin_private_key::GrumpkinPrivateKey, grumpkin_point::GrumpkinPoint};
 
-use dep::std::aes128::aes128_encrypt_slice;
+use dep::std::aes128::aes128_encrypt;
 use crate::keys::point_to_symmetric_key::point_to_symmetric_key;
 
-struct EncryptedLogIncomingBody<Note> {
-    storage_slot: Field,
-    note_type_id: Field,
-    note: Note,
+struct EncryptedLogIncomingBody<T, M> {
+    data: T,
+    plaintext: [u8; M]
 }
 
-impl<Note> EncryptedLogIncomingBody<Note> {
-    pub fn new<N>(
-        storage_slot: Field,
-        note_type_id: Field,
-        note: Note
-    ) -> Self where Note: NoteInterface<N> {
-        Self { storage_slot, note_type_id, note }
-    }
-
-    pub fn compute_ciphertext<N>(
-        self,
-        eph_sk: GrumpkinPrivateKey,
-        ivpk_app: GrumpkinPoint
-    ) -> [u8] where Note: NoteInterface<N> {
-        let serialized_note: [Field; N] = self.note.serialize_content();
-
-        let mut buffer_slice: [u8] = &[];
-
-        let storage_slot_bytes = self.storage_slot.to_be_bytes(32);
-        let note_type_id_bytes = self.note_type_id.to_be_bytes(32);
+impl<T, M> EncryptedLogIncomingBody<T, M> {
+    pub fn from_note<N>(note: T, storage_slot: Field) -> Self where T: NoteInterface<N, M> {
+        let mut plaintext = note.to_be_bytes(storage_slot);
+        let storage_slot_bytes = storage_slot.to_be_bytes(32);
 
         for i in 0..32 {
-            buffer_slice = buffer_slice.push_back(storage_slot_bytes[i]);
+            plaintext[i] = storage_slot_bytes[i];
```

Comment:
Should this not be removed now? Seems like we are just overwriting the bytes with the same bytes?

Reply:
🤦 of course...